passwordtype.py

```diff
@@ -0,0 +1,43 @@
+import bcrypt
+from sqlalchemy import types
+
+def encode(s):
+    return s.encode('utf8')
+
+def hashpassword(password):
+    return bcrypt.hashpw(encode(password), bcrypt.gensalt())
+
+class Password(object):
+
+    def __init__(self, hashed):
+        self.hashed = hashed
```

Comment:
Does not take the password text because `process_bin_param` processes the database values in and out; and if I did the hashing here, I was hashing hashes and passwords would verify.